akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -1225,6 +1230,7 @@ private[stream] object Collect {
 
             case Failure(NonFatal(ex)) ⇒
               holder.supervisionDirectiveFor(decider, ex) match {
+                // this shouldn't happen because it would have failed fast for stopping exceptions
```

Comment:
It can happen if we are already running and while we are processing the head element, the next element fails with an exception (for which the supervisor stops). Then we are already running (and looping through the recursive `pushNextIfPossible` call), so that we might reach here before the async callback is executed. But, wow, are these complicated call paths...

Reply:
Ah, didn't think of that. I'll update the comment.